verilog/analysis/checkers/parameter_name_style_rule_test.cc

```diff
@@ -119,31 +119,38 @@ TEST(ParameterNameStyleRuleTest, ConfigurationStyleParameterAssignment) {
 
   // Make sure configuration parameter name matches corresponding style
 
-  {  // parameter:CamelCase, localparam:ALL_CAPS
+  {  // parameter:([A-Z0-9]+[a-z0-9]*)+(_[0-9]+)? (PascalCase),
+     // localparam:[A-Z_0-9]+ (UPPER_SNAKE_CASE)
     const std::initializer_list<LintTestCase> kTestCases = {
         {"package a; parameter int FooBar = 1; endpackage"},
         {"package a; parameter int ", {kToken, "FOO_BAR"}, " = 1; endpackage"},
         {"module a; localparam int ", {kToken, "FooBar"}, " = 1; endmodule"},
         {"module a; localparam int FOO_BAR = 1; endmodule"},
     };
     RunConfiguredLintTestCases<VerilogAnalyzer, ParameterNameStyleRule>(
-        kTestCases, "parameter_style:CamelCase;localparam_style:ALL_CAPS");
+        kTestCases,
+        "parameter_style_regex:([A-Z0-9]+[a-z0-9]*)+(_[0-9]+)?;localparam_"
```

Comment:
This example shows how hard it is to actually use if just the defaults would be needed; so making it possible to set the `parameter_style:CamelCase` will not only be backwards compatible, but also much more usable while still keeping the possibility open to actually use the regex.